RDG does not support [FromQuery] for optional parameters

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

The Request Delegate Generator does not properly handle optional delegate parameters annotated with `[FromQuery]`, as it does not convert from a string to the expected type.

Removing the attribute removes the error.

In the generated code, the last line of code produces the error.
```cs
                    var test_raw = httpContext.Request.Query["test"];
                    var test_temp = test_raw.Count > 0 ? (string?)test_raw : 10;
```

Without the attribute, the RDG generates this code
```cs
                    var test2_raw = test2_RouteOrQueryResolver(httpContext);
                    var test2_temp = test2_raw.Count > 0 ? (string?)test2_raw : null;
```

In both cases, the resulting parameter appears to be properly defaulted to the value prescribed assuming the temp value is null or empty.

### Expected Behavior

A parameter should be allowed to be annotated with [FromQuery] to be explicit around resolution of the parameter. The parameter should also be allowed to be optional with a default.

### Steps To Reproduce

https://github.com/ryanseipp/dotnet-aot-from-query

See Program.cs line 28

### Exceptions (if any)

```
/home/user/Code/dotnet-aot-from-query/DotnetAotFromQuery/Microsoft.AspNetCore.Http.RequestDelegateGenerator/Microsoft.AspNetCore.Http.RequestDelegateGenerator.RequestDelegateGenerator/GeneratedRouteBuilderExtensions.g.cs(209,37): error CS0173: Type of conditional expression cannot be determined because there is no implicit conversion between 'string' and 'int' [/home/user/Code/dotnet-aot-from-query/DotnetAotFromQuery/DotnetAotFromQuery.csproj]
```

### .NET Version

8.0.100

### Anything else?

_No response_

--- a/src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointParameterEmitter.cs
+++ b/src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointParameterEmitter.cs
@@ -33,7 +33,7 @@ internal static class EndpointParameterEmitter
         }
         else if (endpointParameter.IsOptional)
         {
-            codeWriter.WriteLine($"var {endpointParameter.EmitTempArgument()} = {endpointParameter.EmitAssigningCodeResult()}.Count > 0 ? (string?){endpointParameter.EmitAssigningCodeResult()} : {endpointParameter.DefaultValue};");
+            codeWriter.WriteLine($"var {endpointParameter.EmitTempArgument()} = {endpointParameter.EmitAssigningCodeResult()}.Count > 0 ? (string?){endpointParameter.EmitAssigningCodeResult()} : null;");
         }
         else if (endpointParameter.IsStringValues)
         {

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTests.QueryParameters.cs
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTests.QueryParameters.cs
@@ -125,6 +125,20 @@ app.MapGet("/hello", ([FromQuery]string p1, [FromQuery]string p2) => $"{p1} {p2}
         await VerifyAgainstBaselineUsingFile(compilation);
     }
 
+    [Fact]
+    public async Task MapAction_ExplicitParsableParameter_StringReturn()
+    {
+        var (results, compilation) = await RunGeneratorAsync("""
+app.MapGet("/hello", ([FromQuery] int p1 = 10) => $"{p1}");
+""");
+        var endpoint = GetEndpointFromCompilation(compilation);
+
+        var httpContext = CreateHttpContext();
+
+        await endpoint.RequestDelegate(httpContext);
+        await VerifyResponseBodyAsync(httpContext, "10");
+    }
+
     public static object[][] MapAction_ExplicitQueryParam_NameTest_Data
     {
         get

